# Bay Area, CA - Airbnb Data


## Context

Since its inception in 2008, Airbnb has disrupted the hospitality industry by allowing almost anyone to rent out a spare room and host travelers looking for an overnight stay. 

While Airbnb has publicly available data from many locations, **Santa Clara County is of particular interest as it is one of the major counties in the Bay Area and home to many prominent tech companies and startups in the Silicon Valley.**
As such, home prices and cost of living are exceptionally high in this area, as one will see in the data.


## Content

All data is publicly available under the Creative Commons "Public Domain Dedication" license and has been updated as of June 12th, 2020.Provided are metrics that are publicly visible for each listing (e.g. name, description, price, reviews, etc.), but the dataset contains some Airbnb's internal metrics as well (review score accuracy, host acceptance rate, etc.).

## Acknowledgements

Acknowledgments go to Airbnb for their publicly released datasets that are available at this website.




## Inspiration & ideas:

1. Create a price-suggestion model for new Airbnb hosts who might not know the value of their listing.
    - Text mining with statified house price data
2. Can we predict the rating of an Airbnb listing utilizing **NLP** of the description columns?How have Airbnb
3. prices changed over time? Are prices seasonal?
   - Time series analysis?
4. Which areas in the Santa Clara County are most "popular"?
   - Region analysis
5. Which features of an Airbnb listing are important to add to its perceived value?
   - Feature analysis,PCA
   - Correlation analysis
   

## Import  libraries and data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data = pd.read_csv('./data/Airbnb_Listings.csv')
print(data.shape)
data.head()

(7221, 106)


id                         listing_url       scrape_id last_scraped  \
0   4952   https://www.airbnb.com/rooms/4952  20200530151957   2020-05-30   
1  11464  https://www.airbnb.com/rooms/11464  20200530151957   2020-05-31   
2  17884  https://www.airbnb.com/rooms/17884  20200530151957   2020-05-31   
3  21373  https://www.airbnb.com/rooms/21373  20200530151957   2020-05-30   
4  37512  https://www.airbnb.com/rooms/37512  20200530151957   2020-05-31   

                                  name  \
0    Butterfly Inn - Graceful Living !   
1    Deluxe Private Studio-custom int.   
2                 Silicon Valley Suite   
3  Bonsai Garden Inn in Professorville   
4    Private  room - Parking 3 carport   

                                             summary  \
0  Lovely garden setting in a serene and art-fill...   
1  Custom built Studio with exquisite design. Rea...   
2  A guest suite for one or two, in a house in a ...   
3  Room  in gracious home with beautiful garden. ...   
4  We live in a safe community close to public tr...   

                                               space  \
0  Very comfortable Queen bed and small desk in b...   
1  Description A favorite for international corpo...   
2  This is a private suite at the rear of a house...   
3  Bright, garden-facing room in beautiful home. ...   
4  I have a really nice room in a quiet neighborh...   

                                         description experiences_offered  \
0  Lovely garden setting in a serene and art-fill...                none   
1  Custom built Studio with exquisite design. Rea...                none   
2  A guest suite for one or two, in a house in a ...                none   
3  Room  in gracious home with beautiful garden. ...                none   
4  We live in a safe community close to public tr...                none   

                               neighborhood_overview  ... instant_bookable  \
0  Located in Professorville, Palo Alto, one of t...  ...                f   
1  10 min. to Lucky's and Starbucks at El Camino ...  ...                f   
2  This is a very quiet family neighborhood, but ...  ...                f   
3  This room is in an ultra convenient location i...  ...                f   
4  Our community is a safe environment and at nig...  ...                f   

  is_business_travel_ready          cancellation_policy  \
0                        f                     moderate   
1                        f                     moderate   
2                        f  strict_14_with_grace_period   
3                        f                     moderate   
4                        f                     moderate   

  require_guest_profile_picture require_guest_phone_verification  \
0                             f                                f   
1                             f                                t   
2                             f                                f   
3                             f                                f   
4                             t                                t   

   calculated_host_listings_count  \
0                               5   
1                              12   
2                               2   
3                               5   
4                               2   

   calculated_host_listings_count_entire_homes  \
0                                            0   
1                                           12   
2                                            2   
3                                            0   
4                                            0   

  calculated_host_listings_count_private_rooms  \
0                                            5   
1                                            0   
2                                            0   
3                                            5   
4                                            2   

   calculated_host_listings_count_shared_rooms  reviews_per_month  
0                                

In [3]:
data.shape

(7221, 106)

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7221 entries, 0 to 7220
Columns: 106 entries, id to reviews_per_month
dtypes: float64(23), int64(21), object(62)
memory usage: 5.8+ MB


In [5]:
# find missing values and single value columns in data:
del_cols = []

for col in data.columns:
    if data[col].isna().all():
        del_cols.append(col)
    elif data[col].nunique() == 1:
        del_cols.append(col)

In [6]:
del_cols

['scrape_id',
 'experiences_offered',
 'thumbnail_url',
 'medium_url',
 'xl_picture_url',
 'neighbourhood_group_cleansed',
 'country_code',
 'country',
 'has_availability',
 'requires_license',
 'license',
 'is_business_travel_ready']

In [7]:
data = data.drop(del_cols, axis=1)

In [8]:
data['id'].isna().sum()

0

In [9]:
# delete url column:
urls = [
    'listing_url', 'picture_url', 'host_url', 'host_thumbnail_url',
    'host_picture_url'
]


In [10]:
data = data.drop(urls, axis=1)

In [11]:
data['property_type'].unique()

array(['Villa', 'Apartment', 'Guest suite', 'Bungalow', 'House',
       'Guesthouse', 'Loft', 'Other', 'Condominium', 'Townhouse',
       'Cottage', 'Bed and breakfast', 'Cabin', 'Camper/RV', 'Tiny house',
       'Serviced apartment', 'Treehouse', 'Tent', 'Train', 'Barn', 'Yurt',
       'Boutique hotel', 'Lighthouse', 'Farm stay', 'Campsite',
       'Earth house', 'Aparthotel', 'Chalet'], dtype=object)

In [12]:
data['property_type'].value_counts()

House                 3715
Apartment             1152
Serviced apartment     504
Townhouse              429
Guest suite            363
Guesthouse             318
Condominium            303
Villa                  129
Bungalow               115
Loft                    41
Cottage                 29
Camper/RV               27
Boutique hotel          23
Tiny house              16
Other                   13
Bed and breakfast       12
Cabin                    7
Farm stay                6
Tent                     5
Treehouse                3
Yurt                     3
Barn                     2
Train                    1
Lighthouse               1
Campsite                 1
Earth house              1
Aparthotel               1
Chalet                   1
Name: property_type, dtype: int64

In [13]:
data['room_type'].unique()

array(['Private room', 'Entire home/apt', 'Shared room', 'Hotel room'],
      dtype=object)

In [14]:
accomo_type = data['accommodates'].unique()
accomo_type.sort()
accomo_type

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16])

In [15]:
data['bed_type'].unique()

array(['Real Bed', 'Futon', 'Pull-out Sofa', 'Airbed', 'Couch'],
      dtype=object)

In [16]:
data['jurisdiction_names'].unique()

array(['{"PALO ALTO"}', '{"SANTA CLARA"}', '{"Mountain View"," CA"}',
       '{"SAN JOSE"}', '{Cupertino," CA"}', '{Sunnyvale," CA"}', nan,
       '{Milpitas," CA"}', '{"Morgan Hill"," CA"}', '{"Los Gatos"," CA"}',
       '{"San Benito County"," CA"}'], dtype=object)

In [17]:
data['jurisdiction_names'].isna().sum()

979

In [18]:
data['cancellation_policy'].unique()

array(['moderate', 'strict_14_with_grace_period', 'flexible',
       'super_strict_60'], dtype=object)

In [19]:
data['city'].unique()

array(['Palo Alto', 'Santa Clara', 'Mountain View', 'San Jose',
       'Cupertino', 'Sunnyvale', 'Campbell', 'Milpitas', 'Saratoga',
       'Morgan Hill', 'Los Altos', 'Los Gatos', 'Menlo Park',
       'Los Altos Hills', 'Monte Sereno', 'Hollister', 'Gilroy',
       'San Jose ', 'San Martin', 'Stanford', 'Santa Clara County',
       'Sunnyvale ', 'Palo Alto ', 'Milpitas ', 'Los Gatos ', 'san jose',
       'Mountain View ', 'Danville ', nan, 'Campbell ', '洛斯阿尔托斯',
       'santa clara', 'Fremont', '圣何塞', 'Watsonville'], dtype=object)

In [20]:
data['state'].unique() # to drop

array(['CA', 'Ca', 'ca', nan, 'California '], dtype=object)

In [21]:
data['zipcode'].isna().sum()

55

In [22]:
for col in data.columns:
    list_unique = (col, len(data[col].unique()))
    print(list_unique)

('id', 7221)
('last_scraped', 2)
('name', 7078)
('summary', 6052)
('space', 4403)
('description', 6440)
('neighborhood_overview', 3509)
('notes', 2747)
('transit', 3200)
('access', 3378)
('interaction', 3166)
('house_rules', 3718)
('host_id', 3533)
('host_name', 2103)
('host_since', 2085)
('host_location', 242)
('host_about', 2030)
('host_response_time', 5)
('host_response_rate', 44)
('host_acceptance_rate', 78)
('host_is_superhost', 3)
('host_neighbourhood', 145)
('host_listings_count', 62)
('host_total_listings_count', 62)
('host_verifications', 240)
('host_has_profile_pic', 3)
('host_identity_verified', 3)
('street', 37)
('neighbourhood', 37)
('neighbourhood_cleansed', 16)
('city', 35)
('state', 5)
('zipcode', 83)
('market', 5)
('smart_location', 37)
('latitude', 5882)
('longitude', 6357)
('is_location_exact', 2)
('property_type', 28)
('room_type', 4)
('accommodates', 16)
('bathrooms', 16)
('bedrooms', 11)
('beds', 21)
('bed_type', 5)
('amenities', 6396)
('square_feet', 25)
('price'

In [23]:
# find binary columns:
for col in data.columns:
    if data[col].nunique() == 2:
        print(col)

last_scraped
host_is_superhost
host_has_profile_pic
host_identity_verified
is_location_exact
calendar_last_scraped
instant_bookable
require_guest_profile_picture
require_guest_phone_verification


In [24]:
for col in data.columns:
    if data[col].nunique() == 3:
        print(col)

In [25]:
for col in data.columns:
    if data[col].nunique() == 4:
        print(col)

host_response_time
state
market
room_type
cancellation_policy


In [26]:
for col in data.columns:
    if data[col].nunique() == 5:
        print(col)

bed_type


In [27]:
for col, n in zip(data.columns, data.nunique()):
    if n <10 and n > 1:
        print(col,n)

last_scraped 2
host_response_time 4
host_is_superhost 2
host_has_profile_pic 2
host_identity_verified 2
state 4
market 4
is_location_exact 2
room_type 4
bed_type 5
calendar_last_scraped 2
review_scores_accuracy 8
review_scores_cleanliness 8
review_scores_checkin 8
review_scores_communication 8
review_scores_location 7
review_scores_value 8
instant_bookable 2
cancellation_policy 4
require_guest_profile_picture 2
require_guest_phone_verification 2


In [28]:
data_null = data.isnull().sum()

In [29]:
data_null.sort_values(ascending=False)

square_feet               7192
weekly_price              6647
monthly_price             6629
notes                     3529
access                    3117
                          ... 
maximum_minimum_nights       0
minimum_maximum_nights       0
maximum_maximum_nights       0
minimum_nights_avg_ntm       0
bed_type                     0
Length: 89, dtype: int64

In [30]:
# drop the columns with null values over 50% and state, zipcode

data = data.drop(['square_feet', 'weekly_price', 'monthly_price','zipcode','state'], axis=1)

In [59]:
numeric_features = data.select_dtypes(include=['int64', 'float64'])
nominal_features = data.select_dtypes(include=['object'])

In [60]:
numeric_features

id    host_id  host_listings_count  host_total_listings_count  \
0         4952       7054                  5.0                        5.0   
1        11464      42458                 13.0                       13.0   
2        17884      69023                  2.0                        2.0   
3        21373       7054                  5.0                        5.0   
4        37512     161741                  2.0                        2.0   
...        ...        ...                  ...                        ...   
7216  43567384    5974627                  2.0                        2.0   
7217  43568844  347799221                  1.0                        1.0   
7218  43579879  341673005                  0.0                        0.0   
7219  43580120   31314220                  0.0                        0.0   
7220  43591340    6202502                 71.0                       71.0   

      latitude  longitude  accommodates  bathrooms  bedrooms  beds  ...  \
0     37.43972 -122.15672             2        2.0       1.0   1.0  ...   
1     37.34342 -121.98829             1        1.0       NaN   1.0  ...   
2     37.38612 -122.07357             2        1.0       1.0   1.0  ...   
3     37.44001 -122.15667             1        2.0       1.0   1.0  ...   
4     37.30598 -121.84036             2        1.0       1.0   1.0  ...   
...        ...        ...           ...        ...       ...   ...  ...   
7216  37.33794 -121.87189             1        1.0       0.0   1.0  ...   
7217  37.35473 -121.99940             4        1.0       1.0   2.0  ...   
7218  37.39309 -122.02541             1        1.0       1.0   1.0  ...   
7219  37.31209 -121.92103             1        1.0       1.0   1.0  ...   
7220  37.24502 -121.87745             4        1.0       2.0   2.0  ...   

      review_scores_cleanliness  review_scores_checkin  \
0                          10.0                   10.0   
1                          10.0                   10.0   
2                          10.0                   10.0   
3                          10.0                   10.0   
4                          10.0                   10.0   
...                         ...                    ...   
7216                        NaN                    NaN   
7217                        NaN                    NaN   
7218                        NaN                    NaN   
7219                        NaN                    NaN   
7220                        NaN                    NaN   

      review_scores_communication  review_scores_location  \
0                            10.0                    10.0   
1                             9.0                     9.0   
2                            10.0                    10.0   
3                            10.0                    10.0   
4                            10.0                     9.0   
...                           ...                     ...   
7216                          NaN                     NaN   
7217                          NaN                     NaN   
7218                          NaN                     NaN   
7219                          NaN                     NaN   
7220                          NaN                     NaN   

      review_scores_value  calculated_host_listings_count  \
0                    10.0                               5   
1                     9.0                              12   
2                    10.0                               2   
3                    10.0                               5   
4                     9.0                               2   
...                   ...                             ...   
7216                  NaN                               2   
7217                  NaN                               1   
7218                  NaN                               1   
7219                  NaN                               1   
7220                  NaN                              21   

      calculated_host_l

In [62]:
# id？ host ID？

In [67]:
numeric_features.columns

Index(['id', 'host_id', 'host_listings_count', 'host_total_listings_count',
       'latitude', 'longitude', 'accommodates', 'bathrooms', 'bedrooms',
       'beds', 'guests_included', 'minimum_nights', 'maximum_nights',
       'minimum_minimum_nights', 'maximum_minimum_nights',
       'minimum_maximum_nights', 'maximum_maximum_nights',
       'minimum_nights_avg_ntm', 'maximum_nights_avg_ntm', 'availability_30',
       'availability_60', 'availability_90', 'availability_365',
       'number_of_reviews', 'number_of_reviews_ltm', 'review_scores_rating',
       'review_scores_accuracy', 'review_scores_cleanliness',
       'review_scores_checkin', 'review_scores_communication',
       'review_scores_location', 'review_scores_value',
       'calculated_host_listings_count',
       'calculated_host_listings_count_entire_homes',
       'calculated_host_listings_count_private_rooms',
       'calculated_host_listings_count_shared_rooms', 'reviews_per_month'],
      dtype='object')

In [111]:
nights_range = numeric_features.iloc[:,11:17]
nights_range

minimum_nights  maximum_nights  minimum_minimum_nights  \
0                  2             365                       2   
1                 14             365                      14   
2                 30             180                      30   
3                  2             730                       2   
4                  1               7                       1   
...              ...             ...                     ...   
7216               3            1125                       3   
7217              30            1125                      30   
7218              30            1125                      30   
7219              14             365                      14   
7220               1            1125                       1   

      maximum_minimum_nights  minimum_maximum_nights  maximum_maximum_nights  
0                          2                     365                     365  
1                         14                     365                     365  
2                         30                     180                     180  
3                          2                     730                     730  
4                          1                       7                       7  
...                      ...                     ...                     ...  
7216                       3                    1125                    1125  
7217                      30                    1125                    1125  
7218                      30                    1125                    1125  
7219                      14                    1125                    1125  
7220                       1                    1125                    1125  

[7221 rows x 6 columns]

In [122]:
(nights_range['maximum_maximum_nights'] ==  nights_range['minimum_maximum_nights']).value_counts()
# can we drop these two columns?

True     7178
False      43
dtype: int64

In [61]:
nominal_features

last_scraped                                              name  \
0      2020-05-30                 Butterfly Inn - Graceful Living !   
1      2020-05-31                 Deluxe Private Studio-custom int.   
2      2020-05-31                              Silicon Valley Suite   
3      2020-05-30               Bonsai Garden Inn in Professorville   
4      2020-05-31                 Private  room - Parking 3 carport   
...           ...                                               ...   
7216   2020-05-31          Clean & Cozy Studio Naglee Park San Jose   
7217   2020-05-31  Sophisticated, elegant, remodeled and furnished.   
7218   2020-05-31                           Sunny room in Sunnyvale   
7219   2020-05-30                         Affordable Rose Glen Home   
7220   2020-05-31                             Cozy San Jose Two Bed   

                                                summary  \
0     Lovely garden setting in a serene and art-fill...   
1     Custom built Studio with exquisite design. Rea...   
2     A guest suite for one or two, in a house in a ...   
3     Room  in gracious home with beautiful garden. ...   
4     We live in a safe community close to public tr...   
...                                                 ...   
7216  Clean and Cozy Studio in the heart of beautifu...   
7217  Living Room Furniture: L-shape sopha with coff...   
7218  Sunny room in a single family house of young p...   
7219  Come stay in a comfortable two story home in W...   
7220  Beautifully recently remodeled San Jose two be...   

                                                  space  \
0     Very comfortable Queen bed and small desk in b...   
1     Description A favorite for international corpo...   
2     This is a private suite at the rear of a house...   
3     Bright, garden-facing room in beautiful home. ...   
4     I have a really nice room in a quiet neighborh...   
...                                                 ...   
7216  Stay at this clean and affordable studio with ...   
7217  Sophisticated, elegant, spacious, completely r...   
7218                                                NaN   
7219  -no smoking or drugs allowed on the property -...   
7220  Close to fabulous local hikes (Big Basin State...   

                                            description  \
0     Lovely garden setting in a serene and art-fill...   
1     Custom built Studio with exquisite design. Rea...   
2     A guest suite for one or two, in a house in a ...   
3     Room  in gracious home with beautiful garden. ...   
4     We live in a safe community close to public tr...   
...                                                 ...   
7216  Clean and Cozy Studio in the heart of beautifu...   
7217  Living Room Furniture: L-shape sopha with coff...   
7218  Sunny room in a single family house of young p...   
7219  Come stay in a comfortable two story home in W...   
7220  Beautifully recently remodeled San Jose two be...   

                                  neighborhood_overview  \
0     Located in Professorville, Palo Alto, one of t...   
1     10 min. to Lucky's and Starbucks at El Camino ...   
2     This is a very quiet family neighborhood, but ...   
3     This room is in an ultra convenient location i...   
4     Our community is a safe environment and at nig...   
...                                                 ...   
7216  We are walking distance to SJUC and all the sh...   
7217  Central location for Silicon Valley, 1.5 ml fr...   
7218  in the heart of the Silicon Valley on a quiet ...   
7219  Rose Glen is a quiet, residential neighborhood...   
7220                                                NaN   

                                                  notes  \
0     Housekeeping every Monday, leave your door ope...   
1     Pet Policy: - Well Behaved pet up to 25 lb. of...   
2     I cannot accommodate cats, sorry. I can usuall...   
3     The family room has a flat panel tv and desk f...   
4     Please remember

In [69]:
nominal_features.columns

Index(['last_scraped', 'name', 'summary', 'space', 'description',
       'neighborhood_overview', 'notes', 'transit', 'access', 'interaction',
       'house_rules', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_neighbourhood', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'street',
       'neighbourhood', 'neighbourhood_cleansed', 'city', 'market',
       'smart_location', 'is_location_exact', 'property_type', 'room_type',
       'bed_type', 'amenities', 'price', 'security_deposit', 'cleaning_fee',
       'extra_people', 'calendar_updated', 'calendar_last_scraped',
       'first_review', 'last_review', 'jurisdiction_names', 'instant_bookable',
       'cancellation_policy', 'require_guest_profile_picture',
       'require_guest_phone_verification'],
      dtype='object')

What we got now:

- Pre data cleaning
- Neighbors does not match the neighbors in data
  - sub data by region?
- Location :  clean later
  -  ('latitude', 5882)
  -  ('longitude', 6357)

 

In [32]:
# Calm down, can you do more data cleaning & preprocessing?

In [33]:
from collections import Counter

results = Counter()

In [34]:
amenities = data['amenities']

In [35]:
amenities.str.strip('{}')\
               .str.replace('"', '')\
               .str.lstrip('\"')\
               .str.rstrip('\"')\
               .str.split(',')\
               .apply(results.update)

0       None
1       None
2       None
3       None
4       None
        ... 
7216    None
7217    None
7218    None
7219    None
7220    None
Name: amenities, Length: 7221, dtype: object

In [36]:
results

Counter({'TV': 5203,
         'Cable TV': 2228,
         'Internet': 1838,
         'Wifi': 7071,
         'Kitchen': 5890,
         'Heating': 6824,
         'Washer': 5817,
         'Dryer': 5785,
         'Smoke alarm': 6794,
         'Carbon monoxide alarm': 6178,
         'First aid kit': 3668,
         'Safety card': 769,
         'Fire extinguisher': 4040,
         'Essentials': 6904,
         'Shampoo': 6271,
         'Lock on bedroom door': 3677,
         'Hangers': 6480,
         'Hair dryer': 5997,
         'Iron': 5711,
         'Laptop-friendly workspace': 5998,
         'translation missing: en.hosting_amenity_49': 805,
         'translation missing: en.hosting_amenity_50': 934,
         'Private entrance': 3052,
         'Hot water': 5280,
         'Bed linens': 3710,
         'Extra pillows and blankets': 3024,
         'Microwave': 4541,
         'Coffee maker': 3907,
         'Refrigerator': 4561,
         'Dishwasher': 2829,
         'Dishes and silverware': 4139,
  

In [37]:
sub_df = pd.DataFrame(results.most_common(50), columns=['amenities', 'count'])

In [38]:
sub_df['ratio'] = round(sub_df['count'] / len(amenities),4)

In [39]:
sub_df.sort_values(by=['count'], ascending=False)

amenities  count   ratio
0                         Wifi   7071  0.9792
1                   Essentials   6904  0.9561
2                      Heating   6824  0.9450
3                  Smoke alarm   6794  0.9409
4                      Hangers   6480  0.8974
5                      Shampoo   6271  0.8684
6        Carbon monoxide alarm   6178  0.8556
7    Laptop-friendly workspace   5998  0.8306
8                   Hair dryer   5997  0.8305
9                      Kitchen   5890  0.8157
10                      Washer   5817  0.8056
11                       Dryer   5785  0.8011
12                        Iron   5711  0.7909
13                   Hot water   5280  0.7312
14    Free parking on premises   5277  0.7308
15                          TV   5203  0.7205
16            Air conditioning   4565  0.6322
17                Refrigerator   4561  0.6316
18                   Microwave   4541  0.6289
19       Dishes and silverware   4139  0.5732
20         Free street parking   4043  0.5599
21           Fire extinguisher   4040  0.5595
22                Coffee maker   3907  0.5411
23                  Bed linens   3710  0.5138
24        Lock on bedroom door   3677  0.5092
25               First aid kit   3668  0.5080
26                       Stove   3455  0.4785
27              Cooking basics   3447  0.4774
28                        Oven   3434  0.4756
29               Self check-in   3257  0.4510
30            Private entrance   3052  0.4227
31  Extra pillows and blankets   3024  0.4188
32     Long term stays allowed   2851  0.3948
33                  Dishwasher   2829  0.3918
34          Garden or backyard   2437  0.3375
35            Patio or balcony   2291  0.3173
36                    Cable TV   2228  0.3085
37     Luggage dropoff allowed   1839  0.2547
38                    Internet   1838  0.2545
39           Single level home   1512  0.2094
40                        Pool   1437  0.1990
41         Family/kid friendly   1429  0.1979
42            Indoor fireplace   1345  0.1863
43                      Keypad   1286  0.1781
44         Private living room   1203  0.1666
45                     Lockbox   1197  0.1658
46                     Bathtub   1070  0.1482
47                Pets allowed   1040  0.1440
48                   BBQ grill   1028  0.1424
49                         Gym   1008  0.1396

## Fusion with reviews:

In [40]:
reviews = pd.read_csv('./data/reviews.csv')
print(reviews.shape)
reviews.head()

(212513, 2)


listing_id        date
0        4952  2009-08-02
1        4952  2009-09-04
2        4952  2009-10-16
3        4952  2009-12-10
4        4952  2010-06-08

In [41]:
rev_row = reviews['listing_id'].unique()

In [42]:
data_rev = data.loc[data['id'].isin(rev_row)]
data_rev.head()

id last_scraped                                 name  \
0   4952   2020-05-30    Butterfly Inn - Graceful Living !   
1  11464   2020-05-31    Deluxe Private Studio-custom int.   
2  17884   2020-05-31                 Silicon Valley Suite   
3  21373   2020-05-30  Bonsai Garden Inn in Professorville   
4  37512   2020-05-31    Private  room - Parking 3 carport   

                                             summary  \
0  Lovely garden setting in a serene and art-fill...   
1  Custom built Studio with exquisite design. Rea...   
2  A guest suite for one or two, in a house in a ...   
3  Room  in gracious home with beautiful garden. ...   
4  We live in a safe community close to public tr...   

                                               space  \
0  Very comfortable Queen bed and small desk in b...   
1  Description A favorite for international corpo...   
2  This is a private suite at the rear of a house...   
3  Bright, garden-facing room in beautiful home. ...   
4  I have a really nice room in a quiet neighborh...   

                                         description  \
0  Lovely garden setting in a serene and art-fill...   
1  Custom built Studio with exquisite design. Rea...   
2  A guest suite for one or two, in a house in a ...   
3  Room  in gracious home with beautiful garden. ...   
4  We live in a safe community close to public tr...   

                               neighborhood_overview  \
0  Located in Professorville, Palo Alto, one of t...   
1  10 min. to Lucky's and Starbucks at El Camino ...   
2  This is a very quiet family neighborhood, but ...   
3  This room is in an ultra convenient location i...   
4  Our community is a safe environment and at nig...   

                                               notes  \
0  Housekeeping every Monday, leave your door ope...   
1  Pet Policy: - Well Behaved pet up to 25 lb. of...   
2  I cannot accommodate cats, sorry. I can usuall...   
3  The family room has a flat panel tv and desk f...   
4  Please remember that you are in my home and re...   

                                             transit  \
0  Walking distance to Stanford University (30 mi...   
1  Public transportation at Homestead and Pomeroy...   
2  The CalTrain station is in walking distance, a...   
3  The Stanford Shopping Center is 20 minutes wal...   
4  We have a bus stop right outside our community...   

                                              access  ...  \
0  Kitchen, laundry, family/TV room, garden, free...  ...   
1          Complimentary Wifi-internet + Basic Cable  ...   
2  Private outdoor patio. Shared washer/dryer on ...  ...   
3  Kitchen, Laundry, Garden, Family Room with TV,...  ...   
4  The kitchen and all that it offers.  The laund...  ...   

        jurisdiction_names instant_bookable          cancellation_policy  \
0            {"PALO ALTO"}                f                     moderate   
1          {"SANTA CLARA"}                f                     moderate   
2  {"Mountain View"," CA"}                f  strict_14_with_grace_period   
3            {"PALO ALTO"}                f                     moderate   
4             {"SAN JOSE"}                f                     moderate   

  require_guest_profile_picture require_guest_phone_verification  \
0                             f                                f   
1                             f                                t   
2                             f                                f   
3                             f                                f   
4                             t                                t   

  calculated_host_listings_count calculated_host_listings_count_entire_homes  \
0                              5                                           0   
1                             12                                          12   
2                              2                                           2   
3                              5                          

In [43]:
data_rev.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5734 entries, 0 to 7214
Data columns (total 84 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            5734 non-null   int64  
 1   last_scraped                                  5734 non-null   object 
 2   name                                          5734 non-null   object 
 3   summary                                       5563 non-null   object 
 4   space                                         4571 non-null   object 
 5   description                                   5606 non-null   object 
 6   neighborhood_overview                         3884 non-null   object 
 7   notes                                         3191 non-null   object 
 8   transit                                       3618 non-null   object 
 9   access                                        3642 non-null   o

In [44]:
data_rev.describe()

id       host_id  host_listings_count  \
count  5.734000e+03  5.734000e+03          5733.000000   
mean   2.461825e+07  8.428594e+07           102.008373   
std    1.209688e+07  9.037814e+07           405.285405   
min    4.952000e+03  7.054000e+03             0.000000   
25%    1.489002e+07  1.480914e+07             1.000000   
50%    2.598272e+07  4.800549e+07             3.000000   
75%    3.500704e+07  1.351438e+08             9.000000   
max    4.356033e+07  3.477372e+08          2007.000000   

       host_total_listings_count     latitude    longitude  accommodates  \
count                5733.000000  5734.000000  5734.000000   5734.000000   
mean                  102.008373    37.352121  -121.966650      3.115452   
std                   405.285405     0.065509     0.109539      2.365797   
min                     0.000000    36.965600  -122.188680      1.000000   
25%                     1.000000    37.320290  -122.043607      2.000000   
50%                     3.000000    37.357740  -121.961940      2.000000   
75%                     9.000000    37.398320  -121.886337      4.000000   
max                  2007.000000    37.462740  -121.380120     16.000000   

         bathrooms     bedrooms         beds  ...  review_scores_cleanliness  \
count  5734.000000  5729.000000  5707.000000  ...                5649.000000   
mean      1.371992     1.400768     1.816015  ...                   9.580457   
std       0.678344     0.981305     1.535468  ...                   0.848019   
min       0.000000     0.000000     0.000000  ...                   2.000000   
25%       1.000000     1.000000     1.000000  ...                   9.000000   
50%       1.000000     1.000000     1.000000  ...                  10.000000   
75%       1.500000     2.000000     2.000000  ...                  10.000000   
max       8.000000     9.000000    26.000000  ...                  10.000000   

       review_scores_checkin  review_scores_communication  \
count            5648.000000                  5649.000000   
mean                9.818520                     9.796955   
std                 0.643308                     0.691166   
min                 2.000000                     2.000000   
25%                10.000000                    10.000000   
50%                10.000000                    10.000000   
75%                10.000000                    10.000000   
max                10.000000                    10.000000   

       review_scores_location  review_scores_value  \
count             5648.000000          5648.000000   
mean                 9.797805             9.531339   
std                  0.584477             0.837670   
min                  2.000000             2.000000   
25%                 10.000000             9.000000   
50%                 10.000000            10.000000   
75%                 10.000000            10.000000   
max                 10.000000            10.000000   

       calculated_host_listings_count  \
count                      5734.00000   
mean                         26.45361   
std                          83.43773   
min                           1.00000   
25%                           1.00000   
50%                           3.00000   
75%                           8.00000   
max                         411.00000   

       calculated_host_listings_count_entire_homes  \
count                                  5734.000000   
mean                                     22.485699   
std                                      83.821405   
min                                       0.000000   
25%                                       0.000000   
50%                                       1.000000   
75%                                       2.000000   
max                                     411.000000   

       calculated_host_listings_count_private_rooms  \
count                                   5734.000000   
mean                                       2.877224   
std                          

In [45]:
counts = reviews['listing_id'].value_counts()


In [46]:
counts.head()

7476637     489
52786       478
10814836    445
19641513    429
13828514    427
Name: listing_id, dtype: int64

## Fusion with neighborhood data

In [47]:
neighbors = pd.read_csv('./data/neighbourhoods.csv')
print(neighbors.shape)
neighbors.head()

(16, 2)


neighbourhood_group    neighbourhood
0                  NaN         Campbell
1                  NaN        Cupertino
2                  NaN           Gilroy
3                  NaN        Los Altos
4                  NaN  Los Altos Hills

In [48]:
data_neighbor = data['neighbourhood'].unique().tolist()
neighbors_hood = neighbors['neighbourhood'].unique().tolist()

In [49]:
same_locations = []

for neighbor in neighbors_hood:
    if neighbor in data_neighbor:
        same_locations.append(neighbor)

In [50]:
same_locations

['Campbell',
 'Cupertino',
 'Los Altos',
 'Los Altos Hills',
 'Mountain View',
 'Palo Alto',
 'Santa Clara',
 'Sunnyvale']

In [51]:
data_neighbor

['Palo Alto',
 'Santa Clara',
 'Mountain View',
 'South San Jose',
 'Cupertino',
 'Sunnyvale',
 'Downtown',
 'Campbell',
 'West Valley',
 'Edenvale',
 nan,
 'Willow Glen',
 'Central San Jose',
 'Los Altos',
 'Berryessa',
 'Cory',
 'College Park',
 'Alum Rock',
 'Cambrian/Pioneer',
 'North San Jose',
 'Los Altos Hills',
 'Burbank/Del Monte',
 'Evergreen',
 'Newhall/Sherwood',
 'Shasta/Hanchett Park',
 'Rose Garden',
 'Five Wounds/Brookwood Terrace',
 'Naglee Park',
 'Forest/Pruneridge',
 'Alviso',
 'Japantown',
 'Stanford',
 'Delmas Park',
 'Chapman/Morse',
 'Menlo Park',
 'Autumn/Montgomery',
 'Vermont/McKendrie']

In [52]:
neighbors_hood

['Campbell',
 'Cupertino',
 'Gilroy',
 'Los Altos',
 'Los Altos Hills',
 'Los Gatos',
 'Milpitas',
 'Monte Sereno',
 'Morgan Hill',
 'Mountain View',
 'Palo Alto',
 'San Jose',
 'Santa Clara',
 'Saratoga',
 'Sunnyvale',
 'Unincorporated Areas']